src/resources.rs

```diff
@@ -0,0 +1,334 @@
+//! Embedded resource definitions for Substrate development documentation and tools.
+//! These resources serve as an index for agents to know where to find official documentation,
+//! tools, and references for Substrate blockchain development.
+
+use rmcp::model::{Annotations, RawResource, Resource};
+
+struct SubstrateResource {
+    uri: String,
+    name: String,
+    description: String,
+    content: String,
+    priority: f32,
+}
+
+impl From<SubstrateResource> for Resource {
+    fn from(val: SubstrateResource) -> Self {
+        Resource::new(
+            RawResource {
+                uri: val.uri,
+                name: val.name,
+                description: Some(val.description),
+                mime_type: Some("text/markdown".to_string()),
+                size: Some(val.content.len() as u32),
+            },
+            Some(Annotations {
+                audience: None,
+                priority: Some(val.priority),
+                timestamp: None,
+            }),
+        )
+    }
+}
+
+fn resources() -> Vec<SubstrateResource> {
+    vec![
+        SubstrateResource {
+            uri: "substrate:polkadot-docs".to_string(),
+            name: "Polkadot Documentation Hub".to_string(),
+            description: "Main documentation portal for Polkadot blockchain framework".to_string(),
+            content: r#"# Substrate Documentation Hub
+
+Documentation: https://docs.polkadot.com/
+API Reference: https://paritytech.github.io/polkadot-sdk/master/sc_service/
+
+Core documentation sections:
+- Learn: Concepts, architecture, and theory
+- Build: Practical guides for development
+- Reference: API documentation and glossary
+- Tutorials: Step-by-step learning paths
+"#
```

Comment:
You could use [indoc](https://docs.rs/indoc/latest/indoc/#using-indoc) to help format this better